corpus2alpino/annotators/alpino_client.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+"""
+Wrapper for the Alpino parser.
+"""
+
+import socket
+import re
+import os
+import errno
+
+from subprocess import run, PIPE
+from datetime import date
+from typing import List, Union, cast
+from tempfile import TemporaryDirectory
+
+from corpus2alpino.abstracts import Annotator
+from corpus2alpino.models import Document, MetadataValue
+from corpus2alpino.log import LogSingleton
+
+closing_punctuation = re.compile(r'([^\s])([\.?!])$')
+sentence_id_matcher = re.compile(r'(?<=sentid=")[^"]+(?=")')
+sentence_tag_matcher = re.compile(r'(?<=<sentence)(?![\w-])')
+
+
+def determine_alpino_version(alpino_directory: str):
+    try:
+        version_path = os.path.join(alpino_directory, 'version')
+        version = cast(Union[str, None], open(version_path).read().strip())
+        version_date = cast(Union[date, None], date.fromtimestamp(
+            os.path.getmtime(version_path)))
+    except KeyError:
+        version = None
+        version_date = None
+    return (version, version_date)
+
+
+class AlpinoServerClient:
+    """
+    Wrapper for connecting to an Alpino parser server.
+    """
+
+    def __init__(self, host: str, port: int):
+        self.host = host
+        self.port = port
+
+        self.prefix_id = True
+        self.write_id = False
+        parsed = self.parse_line("hallo wereld !", '42')
+        if '"42|hallo"' in parsed:
+            self.prefix_id = False  # Add it ourselves
+            parsed = self.parse_line("hallo wereld !", '42')
+            if not '"hallo"' in parsed:
+                raise Exception("Alpino has unsupported sentence ID behavior")
+
+        # validate that the match can be found
+        match = sentence_id_matcher.search(parsed)
+        if not match:
+            # no sentence ID added, add it ourselves
+            self.write_id = True
+        else:
+            if self.prefix_id and match.group(0) != "42":
+                raise Exception(
+                    "Unexpected sentence id: {0} instead of 42".format(match.group(0)))
+
+        # detect version
+        if self.host == 'localhost':
```

Comment:
This check makes the program fail when running Alpino on Docker (as I currently am). (as I don't set ALPINO_HOME env).